dotCMS/src/main/java/com/dotmarketing/business/web/HostWebAPI.java

```diff
@@ -28,7 +31,28 @@
 	
 	public Host getCurrentHost(HttpServletRequest req) throws DotDataException, DotSecurityException, PortalException, SystemException;
 
-	public Host getCurrentHost(final HttpServletRequest request, final PageMode mode) throws DotDataException, DotSecurityException, PortalException, SystemException;
+	/**
+	 * Return the current Host taking the host id or name from any of the follow source:
+	 *
+	 * - Current host id set by 'host_id' {@link HttpServletRequest} parameter, just in case the log in user is a backend user.
+	 * - Current host name set by 'Host' {@link HttpServletRequest} parameter
+	 * - Current {@link Host} object set by {@link WebKeys#CURRENT_HOST} {@link HttpServletRequest} attribute.
+	 * - Current {@link Host} object set by {@link WebKeys#CURRENT_HOST} {@link HttpSession} attribute,
+	 * - Current host id set by {@link WebKeys#CMS_SELECTED_HOST_ID} {@link HttpSession} attribute,
+	 * 	 just in case the log in user is a backend user..
+	 *   just in case the log in user is a backend user..
+	 * - Current host name get by {@link HttpServletRequest#getServerName()}
+	 * - Default Host
+	 *
+	 * @param request current {@link HttpServletRequest}
+	 * @param user User current login user
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 * @throws PortalException
+	 * @throws SystemException
+	 */
+	public Host getCurrentHost(final HttpServletRequest request, final User user) throws DotDataException, DotSecurityException;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary modifier 'public' on method 'getCurrentHost': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4812018)